### ISSUE TITLE
Repo: New internal lint rule: prefer .pos over .getFullStart() for TS nodes?

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#8227]
### Suggestion

TypeScript's AST nodes come with three fields to retrieve their starting position:

* `pos: number`
* `getFullStart(): number`
* `getStart(): number`

`getStart()` returns a starting position that includes comments, while `getFullStart()` generally returns `this.pos` in the TypeScript `src/services/services.ts` source I've seen. But I'm not confident that there's no a functional difference for us between `pos` and `getFullStart`. ... is there any?

* If there is no difference, we should lint to enforce one for consistency IMO.
* If there is a difference, then this issue should be closed with an explanation of the difference.

Edit: per https://discord.com/channels/508357248330760243/640177429775777792/1194002057858916483, I'd flipped `getFullStart()` and `getStart()`. Edited.

https://github.com/microsoft/TypeScript/blob/ca7a3af5e607a6c8589f5f600d6c4960a8bc07d4/src/services/services.ts#L564-L570

```ts
    public getFullStart(): number {
        return this.pos;
    }

    public getEnd(): number {
        return this.end;
    }
```

**Comment by JoshuaKGoldberg** [comment]
Actually, I also wonder: since `.getStart()` performs a full `getTokenPosOfNode`, is the performance different significant enough that we'd want to lint for `.pos`/`.getFullStart()`?

**Comment by JoshuaKGoldberg** [comment]
Waiting on resolution to https://github.com/microsoft/TypeScript/pull/57101.

**Comment by JoshuaKGoldberg** [comment]
The TS PR didn't go anywhere and we haven't seen this be an issue since posting. Closing out as not worth the surprisingly large time investment.
